File: 03_hash/md5/base64url/main.java
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

class Main {
    public static void main(String[] args) throws NoSuchAlgorithmException {
        String text = "Hello world, mac was here";
        String hash = Base64.getUrlEncoder().withoutPadding().encodeToString(MessageDigest.getInstance("MD5").digest(text.getBytes(StandardCharsets.UTF_8)));
        System.out.println(hash);
    }
}

File: 03_hash/md5/base64url/main.go
package main

import (
	"crypto/md5"
	"encoding/base64"
	"fmt"
)

func main() {
	text := "Hello world, mac was here"
	hasher := md5.New()
	hasher.Write([]byte(text))
	hash := base64.RawURLEncoding.EncodeToString(hasher.Sum(nil))
	fmt.Println(hash)
}
